File: 190_reverse_bits.py
class Solution:
    def reverseBits(self, n: int) -> int:
        result = 0
        for i in range(32):
            result <<= 1
            result |= n & 1
            n >>= 1
        return result

n = 43261596
solution = Solution()
print(solution.reverseBits(n))  # Output: 964176192

File: 190_reverse_bits.cpp
# include <iostream>
using namespace std;


class Solution {
public:
    int reverseBits(int n) {
        int res = 0;
        for (int i=0; i<32; i++) {
            res <<= 1;
            res |= (n & 1);
            n >>= 1;
        }
        return res;
    }
};

int main(){
    int n = 43261596;
    Solution sol;
    int result = sol.reverseBits(n);
    cout << result << endl;
}